launch-scripts/launch_boot_tests_gem5_20.py

```diff
@@ -120,13 +133,16 @@
     boot_types = ['init', 'systemd']
     num_cpus = ['1', '2', '4', '8']
     cpu_types = ['kvm', 'atomic', 'simple', 'o3']
-    mem_types = ['classic', 'MI_example', 'MESI_Two_Level']
+    mem_types = ['classic', 'MI_example', 'MESI_Two_Level', 'MOESI_CMP_directory']
 
     def createRun(linux, boot_type, cpu, num_cpu, mem):
 
         if mem == 'MESI_Two_Level':
             binary_gem5 = 'gem5/build/X86_MESI_Two_Level/gem5.opt'
             artifact_gem5 = gem5_binary_MESI_Two_Level
+        elif mem == 'MOESI_CMP_directory':
+            binary_gem5 = 'gem5/build/MOESI_CMP_directory/gem5.opt'
+            artifact_gem5 = gem5_binary_MOESI_CMP_directory
         else:
             binary_gem5 = 'gem5/build/X86/gem5.opt'
             artifact_gem5 = gem5_binary
```

Comment:
How about we remove some of this duplication (e.g., removing the `binary_gem5=` lines) with the following

```suggestion
            artifact_gem5 = gem5_binary
        binary_gem5 = artifact_gem5.path
```

Reply:
Thanks, I'll move this along with a few minor improvements to a separate patch.